core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/WorkspaceMigrationActorSpec.scala

```diff
@@ -0,0 +1,731 @@
+import org.scalatest.concurrent.Eventually
+import org.scalatest.flatspec.AnyFlatSpecLike
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.{Assertion, BeforeAndAfterAll, OptionValues}
+import slick.jdbc.MySQLProfile.api._
+
+import java.sql.{SQLException, Timestamp}
+import java.util.UUID
+import scala.language.postfixOps
+
+class WorkspaceMigrationActorSpec
+  extends AnyFlatSpecLike
+    with BeforeAndAfterAll
+    with Matchers
+    with Eventually
+    with OptionValues {
+
+  val testKit: ActorTestKit = ActorTestKit()
+  implicit val logger = new ConsoleLogger("unit_test", LogLevel(false, false, true, true))
+  implicit val ec = IORuntime.global.compute
+  implicit val timestampOrdering = new Ordering[Timestamp] {
+    override def compare(x: Timestamp, y: Timestamp): Int = x.compareTo(y)
+  }
+
+  // This is a horrible hack to avoid refactoring the tangled mess in the WorkspaceServiceSpec.
+  val spec = new WorkspaceServiceSpec()
+
+
+  def runMigrationTest(test: MigrateAction[Assertion]): Assertion =
+    spec.withTestDataServices { services =>
+      test.run {
+        MigrationDeps(
+          services.slickDataSource,
+          GoogleProject("fake-google-project"),
+          services.workspaceService,
+          MockStorageService(),
+          MockStorageTransferService()
+        )
+      }
+        .value
+        .unsafeRunSync
+        .getOrElse(throw new AssertionError("The test exited prematurely."))
+    }
+
+
+  case class MockStorageTransferService() extends MockGoogleStorageTransferService[IO] {
+    override def createTransferJob(jobName: JobName, jobDescription: String, projectToBill: GoogleProject, originBucket: GcsBucketName, destinationBucket: GcsBucketName, schedule: JobTransferSchedule): IO[TransferJob] =
+      IO.pure {
+        TransferJob.newBuilder
+          .setName(s"${jobName}")
+          .setDescription(jobDescription)
+          .setProjectId(s"${projectToBill}")
+          .build
+      }
+
+    override def listTransferOperations(jobName: JobName, project: GoogleProject): IO[Seq[Operation]] =
+      IO.pure {
+        Seq(Operation.newBuilder.setDone(true).build)
+      }
+  }
+
+
+  case class MockStorageService() extends MockGoogleStorageService[IO] {
+    override def deleteBucket(googleProject: GoogleProject, bucketName: GcsBucketName, isRecursive: Boolean, bucketSourceOptions: List[Storage.BucketSourceOption], traceId: Option[TraceId], retryConfig: RetryConfig): fs2.Stream[IO, Boolean] =
+      fs2.Stream.emit(true)
+
+    override def insertBucket(googleProject: GoogleProject, bucketName: GcsBucketName, acl: Option[NonEmptyList[Acl]], labels: Map[String, String], traceId: Option[TraceId], bucketPolicyOnlyEnabled: Boolean, logBucket: Option[GcsBucketName], retryConfig: RetryConfig, location: Option[String]): fs2.Stream[IO, Unit] =
+      fs2.Stream.emit(())
+  }
+
+
+  def getAttempt(workspaceUuid: UUID): ReadWriteAction[Option[WorkspaceMigration]] =
```

Comment:
Could we use a `getMigrationAttempt` method from the real actor for this?

Reply:
The actor code doesn't really need/use this functionality